src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidatorTest.java

```diff
@@ -213,88 +199,196 @@ void testErrorReportedWhenCeaseDateFieldIsNotNullAndBosStillAssociatedWithTrust(
+        trustDataDtoList.get(0).setUnableToObtainAllTrustInfo(false);
 
-        Errors errors = trustDetailsValidator.validate(overseasEntitySubmissionDto, new Errors(), LOGGING_CONTEXT, false);
+        Errors errors = trustDetailsValidator.validate(overseasEntitySubmissionDto, new Errors(), LOGGING_CONTEXT, true);
 
         assertFalse(errors.hasErrors());
     }
 
+
     @Test
-    void testNoValidationErrorReportedWhenCeaseDateIsToday() {
-        disassociateBosFromTrust();
+    void testErrorReportedWhenTrustInvolvedIsNull() {
+        trustDataDtoList.get(0).setTrustInvolvedInOverseasEntity(null);
+        Errors errors = trustDetailsValidator.validate(overseasEntitySubmissionDto, new Errors(), LOGGING_CONTEXT, true);
+        String qualifiedFieldName = getQualifiedFieldName(OverseasEntitySubmissionDto.TRUST_DATA, TrustDataDto.IS_TRUST_INVOLVED_IN_OE);
+        String validationMessage = ValidationMessages.NOT_NULL_ERROR_MESSAGE.replace("%s", qualifiedFieldName);
+        assertError(qualifiedFieldName, validationMessage, errors);
+    }
 
-        trustDataDtoList.get(0).setCeasedDate(LocalDate.now());
-        trustDataDtoList.get(0).setCreationDate(LocalDate.of(1970,1, 1));
 
+    @Test
+    void testErrorReportedWhenTrustIsInvolvedInOverseasEntityAndNoBeneficialOwnersDueToNullTrustIds() {
+        overseasEntitySubmissionDto.getBeneficialOwnersCorporate().get(0).setTrustIds(null);
+        overseasEntitySubmissionDto.getBeneficialOwnersIndividual().get(0).setTrustIds(null);
         Errors errors = trustDetailsValidator.validate(overseasEntitySubmissionDto, new Errors(), LOGGING_CONTEXT, true);
+        String qualifiedFieldName = getQualifiedFieldName(OverseasEntitySubmissionDto.TRUST_DATA, TrustDataDto.IS_TRUST_INVOLVED_IN_OE);
+        String validationMessage = ValidationMessages.INVOLVED_IN_TRUST_WITHOUT_BENEFICIAL_OWNERS_ERROR_MESSAGE;
+        assertError(qualifiedFieldName, validationMessage, errors);
+    }
 
-        assertFalse(errors.hasErrors());
+    @Test
+    void testErrorReportedWhenTrustIsInvolvedInOverseasEntityAndNoBeneficialOwnersDueToEmptyTrustIds() {
+        overseasEntitySubmissionDto.getBeneficialOwnersCorporate().get(0).setTrustIds(List.of());
+        overseasEntitySubmissionDto.getBeneficialOwnersIndividual().get(0).setTrustIds(List.of());
+        Errors errors = trustDetailsValidator.validate(overseasEntitySubmissionDto, new Errors(), LOGGING_CONTEXT, true);
+        String qualifiedFieldName = getQualifiedFieldName(OverseasEntitySubmissionDto.TRUST_DATA, TrustDataDto.IS_TRUST_INVOLVED_IN_OE);
+        String validationMessage = ValidationMessages.INVOLVED_IN_TRUST_WITHOUT_BENEFICIAL_OWNERS_ERROR_MESSAGE;
+        assertError(qualifiedFieldName, validationMessage, errors);
     }
 
     @Test
-    void testErrorReportedWhenCeaseDateIsBeforeCreationDate() {
-        disassociateBosFromTrust();
+    void testErrorReportedWhenTrustIsInvolvedInOverseasEntityAndNoBeneficialOwnersDueToCeasedDate() {
```

Comment:
Would a better name for this test be `testErrorReportedWhenTrustIsInvolvedInOverseasEntityAndAllBeneficialOwnersAreCeased`? assuming that's what it's testing.

Reply:
Prefer to keep the current name as it follows the "due to" pattern.